### PR TITLE
Agent nomination update

### PR DESCRIPTION
**Pull request by Buckram123**
Not sure if that feels right:
It doesn't rely on total tasks at all

### CODE REVIEW

File: contracts/croncat-agents/src/contract.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
+use croncat_sdk_agents::types::AgentNominationStatus;
```
Comment (TrevorJTClarke) [inline contracts/croncat-agents/src/contract.rs:3]: should be on line 19?

File: contracts/croncat-agents/src/contract.rs [hunk 2]
```diff
@@ -67,6 +68,13 @@ pub fn instantiate(
         CONTRACT_NAME,
         version.unwrap_or_else(|| CONTRACT_VERSION.to_string()),
     )?;
+    AGENT_NOMINATION_STATUS.save(
```
Comment (TrevorJTClarke) [inline contracts/croncat-agents/src/contract.rs:71]: oh nice, thats cleaner

File: contracts/croncat-agents/src/contract.rs [hunk 3]
```diff
@@ -578,53 +588,27 @@ fn get_agent_status(
...
+        .start_height_of_nomination
+        .map_or(0, |start_height| {
+            (block_height - start_height) / cfg.agent_nomination_block_duration as u64
+        });
+    let agents_to_pass = min(agents_by_tasks_created, agents_by_height);
+    if agents_to_pass == 0 {
+        Ok(None)
     } else {
-        0
+        Ok(Some(agents_to_pass - 1))
```
Comment (TrevorJTClarke) [inline contracts/croncat-agents/src/contract.rs:608]: still could be 0 right?

File: packages/croncat-sdk-agents/src/types.rs [hunk 4]
```diff
@@ -2,6 +2,12 @@ use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{Addr, Timestamp};
 use std::fmt;
 
+#[cw_serde]
+pub struct AgentNominationStatus {
```
Comment (TrevorJTClarke) [inline packages/croncat-sdk-agents/src/types.rs:6]: luvvit

File: packages/croncat-sdk-core/src/internal_messages/agents.rs [hunk 5]
```diff
@@ -2,9 +2,7 @@ use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{to_binary, Addr, Binary, CosmosMsg, StdResult, WasmMsg};
 
 #[cw_serde]
-pub struct AgentOnTaskCreated {
-    pub task_hash: String,
-}
+pub struct AgentOnTaskCreated {}
```
Comment (TrevorJTClarke) [inline packages/croncat-sdk-core/src/internal_messages/agents.rs:5]: much better

**Review by TrevorJTClarke — APPROVED** [review]
Soooooooooooo much better, YASS

File: contracts/croncat-agents/src/contract.rs (hunk 3 above)
Comment (Buckram123) [reply contracts/croncat-agents/src/contract.rs:608]: Yeah, if 1 agent to pass it would be 0th index

**Pull request merged**